File: Advent_of_Code_2022/Day 1/solution.py
with open('input.txt', 'r') as f:
    max_cal = 0
    curr_cal = 0
    for line in f.readlines():
        line = line.rstrip()
        if(line == ""):
            max_cal = max(max_cal, curr_cal)
            curr_cal = 0
        else:
            curr_cal += int(line)
    print(max_cal)

File: Advent_of_Code_2022/Day 1/solution.cpp
#include <fstream>
#include <string>
#include <iostream>
#include <algorithm>
using namespace std;



int main() {
    ifstream infile("input.txt");
    string line;
    int curr_cal = 0, max_cal = 0;
    while (getline(infile, line)){
        line.erase(remove(line.begin(), line.end(), '\n'),line.end());
        if(line == ""){
            max_cal = max(max_cal, curr_cal);
            curr_cal = 0;
        }
        else{
            curr_cal += stoi(line);
        }
    }
    cout << max_cal << endl;
}
